sources/packages/api/src/services/atbc/atbc.controller.spec.ts

```diff
@@ -0,0 +1,97 @@
+require("../../../env_setup");
+import { closeDB, setupDB } from "../../testHelpers";
+import { Connection } from "typeorm";
+import * as faker from "faker";
+import { Student, User } from "../../database/entities";
+import { StudentService, ArchiveDbService, ATBCService } from "../../services";
+import { KeycloakConfig } from "../../auth/keycloakConfig";
+import { KeycloakService } from "../auth/keycloak/keycloak.service";
+import { ConfigService } from "../config/config.service";
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import * as request from "supertest";
+import { ATBCCreateClientResponse } from "../../types";
+
+describe("Test student model", () => {
+  const clientId = "student";
+  let connection: Connection;
+  let accesstoken: string;
+  let configService: ConfigService;
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    await KeycloakConfig.load();
+    const token = await KeycloakService.shared.getToken(
+      process.env.E2E_TEST_STUDENT_USERNAME,
+      process.env.E2E_TEST_STUDENT_PASSWORD,
+      clientId,
+    );
+    accesstoken = token.access_token;
+
+    connection = await setupDB();
+    const moduleFixture: TestingModule = await Test.createTestingModule(
+      {},
+    ).compile();
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+  afterAll(async () => {
+    await closeDB();
+  });
+
+  it("should save student model object with user relationship and address jsonb", async () => {
```

Comment:
I believe that the main validation on this test would be the "apply-pd-status". Please consider changing it to something like: "should return an HTTP 200 status when applying for PD and student is valid"